### PR TITLE
Add new PFE functions

### PR DESCRIPTION
**Pull request by mrwyattii**
- Implement commonly used transformations
- Add new activation functions
- Mimic pytorch layers to help users move models from pytorch -> LBANN

### CODE REVIEW

**Review by bvanessen — APPROVED** [review]

File: python/lbann/modules/__init__.py [hunk 1]
```diff
@@ -10,3 +10,7 @@
 from lbann.modules.rnn import LSTMCell, GRU, ChannelwiseGRU
 from lbann.modules.transformer import MultiheadAttention
 from lbann.modules.graph import *
+from lbann.modules.subgraph import *
+from lbann.modules.activations import *
+from lbann.modules.pytorch import *
+from lbann.modules.transformations import *
```
Comment (timmoon10) [inline python/lbann/modules/__init__.py:16]: I prefer being explicit about what we're importing:
```suggestion
from lbann.modules.activations import Gelu, Gelu_approx, Silu
from lbann.modules.pytorch import PytorchMatmul, PytorchLinear, PytorchLayerNorm
from lbann.modules.transformations import Permute, Cumsum
```
We ran into some trouble from blindly importing `lbann.modules.subgraph` (https://github.com/LLNL/lbann/pull/2026).

File: python/lbann/modules/pytorch.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        z = lbann.Reshape(z, dims=str_list(z_shape))
+
+    if return_dims:
+        return z, z_shape
+    return z
+
+
+# Mimics torch.nn.Linear in LBANN
+def PytorchLinear(x, input_shape, hidden_size, weights=[], name="", return_dims=False):
+    need_reshape = len(input_shape) > 2
```
Comment (timmoon10) [inline python/lbann/modules/pytorch.py:36]: I think we need a reshape if the data is 1D, or we should use the basic fully-connected layer.

**Review by timmoon10 — CHANGES_REQUESTED** [review]
I've just copy-pasted my comments from https://github.com/LLNL/lbann/pull/1999.

File: python/lbann/modules/pytorch.py (hunk 2 above)
Comment (mrwyattii) [reply python/lbann/modules/pytorch.py:36]: Added a check for 1D inputs with recent commit.

**Review by timmoon10 — APPROVED** [review]
:shipit:

**Pull request merged**